src/tools/run-make-support/src/command.rs

```diff
@@ -163,7 +163,16 @@ impl CompletedProcess {
     /// Checks that trimmed `stdout` matches trimmed `content`.
     #[track_caller]
     pub fn assert_stdout_equals<S: AsRef<str>>(&self, content: S) -> &Self {
-        assert_eq!(self.stdout_utf8().trim(), content.as_ref().trim());
+        assert_eq!(
+            self.stdout_utf8().trim(),
+            content.as_ref().trim(),
+            "{}",
+            format!(
+                "The stdout \"{}\" did not equal the string \"{}\"",
+                self.stdout_utf8().as_str(),
+                content.as_ref().trim()
+            )
```

Comment:
Suggestion: AFAICT this is already a format string, so I think you can write

```suggestion
            "the stdout \"{}\" did not equal the string \"{}\"",
            self.stdout_utf8().as_str(),
            content.as_ref().trim()
```

but also, I'm thinking what if stdout here is multiline (which is quite common I would think?), ths panic message would be very distorted. I'm thinking maybe it's better if we do something like

```rs
let stdout_trimmed = self.stdout_utf8().trim();
let expected_trimmed = expected.as_ref().trim();
if stdout_trimmed != content_trimmed {
    eprintln!("=== STDOUT (ACTUAL) TRIMMED ===");
    eprintln!("{}", stdout_trimmed);
    eprintln!("=== EXPECTED TRIMMED ===");
    eprintln!("{}", expected_trimmed);
    panic!("trimmed stdout does not equal trimmed expected");
}
```

The rationale I want to propose is that if a run-make test does fail in CI, it should be sufficient to tell from the CI log *what* about this assertion failed. And that being verbose here is very good.

I also think that `expected` is a better term here than `content`.

But this is just what I think might help the test writer to diagnose test failure, what do you think?

Reply:
`assert_not_contains` ironically already has something like this in `lib.rs`

```rs
/// Check that `haystack` does not contain `needle`. Panic otherwise.
#[track_caller]
pub fn assert_not_contains(haystack: &str, needle: &str) {
    if haystack.contains(needle) {
        eprintln!("=== HAYSTACK ===");
        eprintln!("{}", haystack);
        eprintln!("=== NEEDLE ===");
        eprintln!("{}", needle);
        panic!("needle was unexpectedly found in haystack");
    }
}

The logic should be easy to carry over.